### PR TITLE
Bug 1797681: Moved Event nav item from Advanced to Monitoring page as tab

### PR DESCRIPTION
**Pull request by abhi-kn**
**Fixes**: 
JIRA ID: https://issues.redhat.com/browse/ODC-2895
Bugzilla ID: https://bugzilla.redhat.com/show_bug.cgi?id=1797681

**Analysis / Root cause**: 
In draft design, Events tab was not included as part of Monitoring page but later it has been added into final design & hence movement of Event page from Advanced to Monitoring page was missed. 

**Solution Description**: 
- Event page has been moved under Monitoring page as a tab. 
- EventsList component in events.jsx has hardcoded Page title 'Events' but it's not required if rendered as a tab since tab name exists. Hence page title is now driven through props with default title as 'Events'. 
- PageHeading component is modified to include margin for title only when title exists

**Screen shots / Gifs for design review**: 
@openshift/team-devconsole-ux 
EventsTab

**Unit test coverage report**: 
image

**Test setup:** Regular

**Browser conformance**: 

- [x] Chrome
- [ ] Firefox
- [x] Safari
- [ ] Edge

### CODE REVIEW

File: frontend/public/components/events.jsx [hunk 1]
```diff
@@ -227,7 +227,10 @@ export class EventsList extends React.Component {
         <Helmet>
           <title>Events</title>
         </Helmet>
```
Comment (christianvogt) [inline frontend/public/components/events.jsx]: I think this should move to `EventStreamPage` component.

File: frontend/public/components/utils/headings.tsx [hunk 2]
```diff
@@ -119,7 +119,10 @@ export const PageHeading = connectToModel((props: PageHeadingProps) => {
           {badge && <SplitItem>{badge}</SplitItem>}
         </Split>
       )}
-      <h1 className={classNames('co-m-pane__heading', { 'co-m-pane__heading--logo': props.icon })}>
+      <h1
+        className={classNames('co-m-pane__heading', { 'co-m-pane__heading--logo': props.icon })}
+        style={resourceTitle ? {} : { margin: 0 }}
```
Comment (christianvogt) [inline frontend/public/components/utils/headings.tsx]: There's potentially many other things in this component that still requires the margin and you cannot simply check for presence of `resourceTitle`.
You should probably just omit rendering the `h1` altogether if there's no contents to render.

File: frontend/public/components/events.jsx [hunk 3]
```diff
@@ -227,7 +227,10 @@ export class EventsList extends React.Component {
...
-        <PageHeading detail={true} title="Events">
+        <PageHeading
+          detail={true}
+          title={this.props.pageTitle !== undefined ? this.props.pageTitle : 'Events'}
```
Comment (christianvogt) [inline frontend/public/components/events.jsx]: This feels a bit backwards but it works.
I was thinking that perhaps `EventStreamPage` should set the title, and by default this component doesn't use a title. Therefore in `MonitoringEvents` you won't need to specify an empty `pageTitle` prop.
Prop name `title` would be more concise.

File: frontend/public/components/events.jsx (hunk 1 above)
Comment (abhi-kn) [reply frontend/public/components/events.jsx]: Yes. Moved as suggested

File: frontend/public/components/events.jsx (hunk 3 above)
Comment (abhi-kn) [reply frontend/public/components/events.jsx]: Modified `EventStreamPage` to set the title.

File: frontend/public/components/utils/headings.tsx (hunk 2 above)
Comment (abhi-kn) [reply frontend/public/components/utils/headings.tsx]: Added a check & reverted previous change.

**Pull request merged**